### ISSUE TITLE
Staging Review finding: Missing `legend` and multiple H3s on several pages

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#108126]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Disability Benefits Experience Team 2
**Product:** Disability Comp Application (21-526 EZ)
**Feature:** Updated 0781 Flow

## Findings details
**VA.gov Experience Standard - issue:** [User has enough information to complete a task.](https://design.va.gov/about/experience-standards/#user-has-enough-information-to-complete-a-task)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

Throughout the mental health flow, pages have a structure that doesn't quite meet HTML spec.
- The headings, helper text, and inputs are wrapped in a `fieldset` (good!)
- The `fieldset` doesn't have a [`legend`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/legend) to describe its contents (problem!)
- The content of the form page is introduced by a pair of H3 headings that are visually grouped as a single element but are made available to assistive technologies as two separate headings (problem!)

This can lead to potential confusion for screen reader users. For example, when landing on `/event-police-report` focus is sent to the first H3 heading, "VA Form 21-0781" and so that text is announced to the user --- but not the subsequent text "Police report location." If a screen reader user then tabs to the first input, "Police report location" still isn't announced, and the label on the first question reads "Name of the agency that issued the report" without any clear indication of what kind of report this page is about. A user will need to backtrack through the page and hunt around to find the context they missed, which may not always be easy to do when completing an emotionally charged form like this.

<details><summary>Link, screenshot or steps to recreate</summary>

Example of how one page is currently coded:
```
<fieldset class="vads-u-margin-y--2 rjsf-object-field">
   <h3 class="vads-u-font-family--sans vads-u-font-size--base vads-u-font-weight--normal vads-u-margin--0">VA FORM 21-0781 </h3>
   <h3 class="vads-u-font-size--h3 vads-u-color--base vads-u-margin--0">Police report location</h3>
   <p>Knowing the report location will help us get a copy of the report. This will help support your claim.</p>
   <div>
      <va-text-input label="Name of the agency that issued the report" name="root_agency" required="false" uswds="true" value="" type="text" class="rjsf-web-component-field hydrated" show-input-error=""></va-text-input>
   </div>
   ...
</fieldset>
```

</details>

## Recommended action

This should be a pretty straightforward tweak to how the HTML is structured, again using the police report example:
```
<fieldset class="vads-u-margin-y--2 rjsf-object-field">
   <legend>
      <h3>
         <span class="...">VA FORM 21-0781 </span>
         <span class="...">Police report location</span>
      </h3>
   </legend>
   <p>Knowing the report location will help us get a copy of the report. This will help support your claim.</p>
   <div>
      <va-text-input label="Name of the agency that issued the report" name="root_agency" required="false" uswds="true" value="" type="text" class="rjsf-web-component-field hydrated" show-input-error=""></va-text-input>
   </div>
   ...
</fieldset>
```

Each `span` within the `h3` and the `legend` can be styled however you'd like, so no visual changes are needed.

Here's what that gets you:
- Focus can be sent to the single H3, meaning screen reader users will hear "VA Form 21-0781, Police report location" when they land on each page of this flow. That extra context helps a ton!
- By grouping things inside a `legend`, the first time you focus moves into that that `fieldset` screen reader users will hear the H3 text announced --- again, "VA Form 21-0781, Police report location." That accounts for a situation where a user is doing something unusual, eg. shift-tabbing up to the nav then back down to the form content. If for some reason they miss the initial announcement of where they are via focus management, they get it from the `legend` on the `fieldset`.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by lisacapaccioli** [comment]
Not sure if I tested this one properly, someone with more experience might want to test this. I feel like my movement was going all over the place. I tried it a couple of times and just ended up keeping this one last result. It doesn't show every step just my most recent test. 

https://github.com/user-attachments/assets/8c31a040-aee7-4e92-acc8-82b28a05bbbd

**Comment by freeheeling** [comment]
I would consider this complete, but given Lisa's comments of uncertainty perhaps someone else should confirm. Thanks.

**Comment by etanb** [comment]
Fixed with: https://github.com/department-of-veterans-affairs/vets-website/pull/36076
